pkg/rhcos/cache/flock_windows.go

```diff
@@ -0,0 +1,19 @@
+//go:build windows
+// +build windows
+
+package cache
+
+import (
+	"os"
+
+	"golang.org/x/sys/windows"
+)
+
+func flockFile(f *os.File, lock bool) error {
+	if lock {
+		var overlapped windows.Overlapped
+		return windows.LockFileEx(windows.Handle(f.Fd()), windows.LOCKFILE_EXCLUSIVE_LOCK, 0, 1, 0, &overlapped)
+	}
+	var overlapped windows.Overlapped
+	return windows.UnlockFileEx(windows.Handle(f.Fd()), 0, 1, 0, &overlapped)
+}
```

Comment:
This seems like locking the first byte in the file? I guess it works, but should we lock the entire file?

```go
const (
      reserved = 0
      allBytes = ^uint32(0)
  )

  func flockFile(f *os.File, lock bool) error {
      if lock {
          var overlapped windows.Overlapped
          return windows.LockFileEx(windows.Handle(f.Fd()), windows.LOCKFILE_EXCLUSIVE_LOCK, reserved, allBytes, allBytes, &overlapped)
      }
      var overlapped windows.Overlapped
      return windows.UnlockFileEx(windows.Handle(f.Fd()), reserved, allBytes, allBytes, &overlapped)
  }
```

For example: https://tip.golang.org/src/cmd/go/internal/lockedfile/internal/filelock/filelock_windows.go